core/src/test/scala/org/apache/spark/shuffle/sort/SortShuffleManagerSuite.scala

```diff
@@ -131,4 +139,29 @@ class SortShuffleManagerSuite extends SparkFunSuite with Matchers {
     )))
   }
 
+  test("Data could not be cleaned up when unregisterShuffle") {
```

Comment:
nit: `SPARK-34541: Data ...`

Reply:
BTW, this test requires `spark.shuffle.useOldFetchProtocol=true`?

And it's better to test both `true` and `false`.

Reply:
It seems that A and B are the same, both increasing from 0. I don't understand why the protocol should be differentiated on the write side

Reply:
We can run a simple job before our target job to make the `taskAttemptId` starts from 1. e.g., 

`sc.parallelize(1 to 10, 1).count()`.

I tried this way and the issue can be reproduced.

Reply:
OK, thank you. I'll add test later, And I don`t understand why the protocol should be differentiated on the WriteSide, As follow:
`ShuffleMapTask#runTask

   // While we use the old shuffle fetch protocol, we use partitionId as mapId in the
    // ShuffleBlockId construction.
    val mapId = if (SparkEnv.get.conf.get(config.SHUFFLE_USE_OLD_FETCH_PROTOCOL)) {
      partitionId
    } else context.taskAttemptId()
`

Reply:
I roughly remember that's because we want to ensure the unique file name at write size. cc @xuanyuanking

Reply:
updated

Reply:
`at write size` -> `at write side`? :)
Yes, you can check the description in #25620. TL;DR: we need a unique file name to resolve the indeterminate shuffle issue.